src/main/telemetry/sim.c

```diff
@@ -0,0 +1,380 @@
+            if (simResponse[simResponseIndex] == '\r') simResponse[simResponseIndex] = '\0';
+            simResponseIndex = 0; //data ok
+            readSimResponse();
+            break;
+        } else {
+            simResponse[simResponseIndex] = c;
+            simResponseIndex++;
+        }
+    }
+
+    detectAccEvents();
+    detectFailsafe();
+    transmit();
+
+    if (now < sim_t_stateChange)
+        return;
+
+    sim_t_stateChange = now + SIM_AT_COMMAND_DELAY_MS;       // by default, if OK or ERROR not received, wait this long
+    simWaitAfterResponse = false;   // by default, if OK or ERROR received, go to next state immediately.
+    switch (simTelemetryState) {
+        case SIM_STATE_INIT:
+        sendATCommand("AT\r");
+        simTelemetryState = SIM_STATE_INIT2;
+        break;
+        case SIM_STATE_INIT2:
+        sendATCommand("ATE0\r");
+        simTelemetryState = SIM_STATE_INIT_ENTER_PIN;
+        break;
+        case SIM_STATE_INIT_ENTER_PIN:
+        sendATCommand("AT+CPIN=" SIM_PIN "\r");
+        simTelemetryState = SIM_STATE_SET_MODES;
+        break;
+        case SIM_STATE_SET_MODES:
+        sendATCommand("AT+CMGF=1;+CNMI=3,2;+CLIP=1;+CSQ\r");
+        simTelemetryState = SIM_STATE_INIT;
+        sim_t_stateChange = now + SIM_CYCLE_MS;
+        simWaitAfterResponse = true;
+        break;
+        case SIM_STATE_SEND_SMS:
+        sendATCommand("AT+CMGS=\"");
+        sendATCommand((char*)telemetryConfig()->simGroundStationNumber);
+        sendATCommand("\"\r");
+        simTelemetryState = SIM_STATE_SEND_SMS_ENTER_MESSAGE;
+        sim_t_stateChange = now + 100;
+        break;
+        case SIM_STATE_SEND_SMS_ENTER_MESSAGE:
+        sendSMS();
+        simTelemetryState = SIM_STATE_INIT;
+        sim_t_stateChange = now + SIM_CYCLE_MS;
+        simWaitAfterResponse = true;
+        break;
+    }
+}
+
+
+void sendATCommand(const char* command)
+{
+    atCommandStatus = SIM_AT_WAITING_FOR_RESPONSE;
+    int len = strlen((char*)command);
+    if (len >SIM_AT_COMMAND_MAX_SIZE)
+        len = SIM_AT_COMMAND_MAX_SIZE;
+    serialWriteBuf(simPort, (const uint8_t*) command, len);
+}
+
+
+void sendSMS()
+{
+    int32_t lat = 0, lon = 0, alt = 0, gs = 0;
+    int vbat = getBatteryVoltage();
+    int16_t amps = isAmperageConfigured() ? getAmperage() / 10 : 0; // 1 = 100 milliamps
+    int avgSpeed = (int)round(10 * calculateAverageSpeed());
+    uint32_t now = millis();
+
+    if (sensors(SENSOR_GPS)) {
+        lat = gpsSol.llh.lat;
+        lon = gpsSol.llh.lon;
+        gs = gpsSol.groundSpeed / 100;
+    }
+#if defined(USE_NAV)
+    alt = getEstimatedActualPosition(Z); // cm
+#else
+    alt = sensors(SENSOR_GPS) ? gpsSol.llh.alt : 0; // cm
+#endif
+    int len;
+    int32_t E7 = 10000000;
+    // \x1a sends msg, \x1b cancels
+    len = tfp_sprintf((char*)atCommand, "%s%d.%02dV %d.%dA ALT:%ld SPD:%ld/%d.%d DIS:%d/%d SAT:%d SIG:%d %s google.com/maps/place/%ld.%07ld,%ld.%07ld,500m\x1a",
```

Comment:
When I using this google link the point is not appears. Problem is in ",500m" when I remove this it work well.
But I have better idea:
maps.google.com/?q=@%ld.%07ld,%ld.%07ld
This link is better because it work in internet browser google maps and in application google maps even when there is not internet connection (offline).